### PR TITLE
EVM-533 Nightly slack notificatinos

### PR DESCRIPTION
**Pull request by Nemanja0x**
Add nightly build slack notifications:
1. link to actions
2. build details

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

**Review by vcastellm — COMMENTED** [review]
Can we make it a little bit more eye-catching?
image

**Comment by igorcrevar** [comment]
> Can we make it a little bit more eye-catching? image

Screenshot from 2023-04-03 13-34-46

**Review by goran-ethernal — APPROVED** [review]

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -27,4 +27,73 @@ jobs:
...
+                  "text": {
+                    "type": "plain_text",
+                    "text": "Nightly Build"
+                  }
+                },
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "All checks ${{ needs.build.outputs.workflow_output == '' && needs.test.outputs.workflow_output == '' && needs.property.outputs.workflow_output == '' && needs.e2e.outputs.workflow_output == '' && ':white_check_mark:' || ':x: `failed`' }}"
```
Comment (Stefan-Ethernal) [inline .github/workflows/nightly.yml]: Rename this to Status maybe and even bold it (maybe even add a divider section below it).

```suggestion
                    "text": "**Status** ${{ needs.build.outputs.workflow_output == '' && needs.test.outputs.workflow_output == '' && needs.property.outputs.workflow_output == '' && needs.e2e.outputs.workflow_output == '' && ':white_check_mark:' || ':x: `failed`' }}"
```

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -27,4 +27,73 @@ jobs:
...
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "Build ${{ needs.build.outputs.workflow_output == '' && ':white_check_mark:' || ':x: `failed`' }}"
+                  }
+                },
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "Unit tests ${{ needs.test.outputs.workflow_output == '' && ':white_check_mark:' || ':x: `failed`' }}"
```
Comment (Stefan-Ethernal) [inline .github/workflows/nightly.yml:71]: It would be great if we would have URLs to specific GH action which executed the given workflow.

**Review by Stefan-Ethernal — APPROVED** [review]
Generally looks good. Left some suggestions to consider.

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (igorcrevar) [reply .github/workflows/nightly.yml:71]: On the bottom of the output, there is link to results of whole night build workflow
https://github.com/0xPolygon/polygon-edge/actions/runs/4596541745
one can click on any of the boxes in the middle of the screen to see logs of particular sub-workflow
Screenshot from 2023-04-03 15-18-04

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (igorcrevar) [reply .github/workflows/nightly.yml]: fixed

**Review by vcastellm — APPROVED** [review]
🎩

**Pull request merged**